models/common_data_models/shopify__line_item_enhanced.sql

```diff
@@ -0,0 +1,171 @@
+with line_items as (
+
+    select * 
+    from {{ var('shopify_order_line')}}
+
+), orders as ( 
+
+    select * 
+    from {{ var('shopify_order')}}
+
+), product as (
+
+    select *
+    from {{ var('shopify_product')}}
+
+), transactions as (
+
+    select *
+    from {{ var('shopify_transaction')}}
+    where kind = 'capture'
+
+), refund_transactions as (
+
+    select
+        order_id,
+        sum(amount) as total_order_refund_amount
+    from {{ var('shopify_transaction')}}
+    where kind = 'refund' 
+    group by 1
+
+), order_line_refund as (
+
+    select *
+    from {{ var('shopify_order_line_refund')}}
+
+), customer as (
+
+    select *
+    from {{ var('shopify_customer')}}
+
+), enhanced as (
+
+    select
+        li.order_id as header_id,
+        li.order_line_id as line_item_id,
+        li.index as line_item_index,
+        o.created_timestamp as created_at,
+        o.currency as currency,
+        li.fulfillment_status as header_status,
+        li.product_id as product_id,
+        p.title as product_name,
+        t.kind as transaction_type,
+        null as billing_type,
+        p.product_type as product_type,
+        li.quantity as quantity,
+        li.price as unit_amount,
+        o.total_discounts as discount_amount,
```

Comment:
I probably just missed this -- updated to use the more granular

Reply:
@fivetran-joemarkiewicz  Ohh wait, I remember now. I used the `total_discounts` from the `order_id` level because I was thinking of the case where discounts get applied to the whole order, but not at the line item. For example, if it's a _get $20 off $50 order_. I don't think it would get attached to just a line_item. Therefore, by using` li.total_discounts` (the more granular one), we would be miscalculating the total discount at the order level.


Same with using o.total_tax instead of the line_item tax field!

Reply:
That makes sense to me. Thanks for the clarification!